renderer/components/dashboard-projects/dashboard-filter/projects-filter.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ProjectDetails,
+  SearchForm,
+} from '../../../modules/projects/redux/stepper/data.model';
+
+export class ProjectsFilter {
+  private allProjects: ProjectDetails[];
+  constructor(projects: ProjectDetails[]) {
+    this.allProjects = projects;
+  }
+  applyFilter(searchFormState: SearchForm): ProjectDetails[] {
+    const { searchValue, filterValue } = searchFormState;
+    const filterBasedOn = this.selectedFilterOption(filterValue);
+
+    switch (filterBasedOn) {
+      case 'domain':
+        if (!searchValue) {
+          return this.filterProjectsWithTechnologies(
+            filterBasedOn,
+            filterValue
+          );
+        } else {
+          return this.filterProjectsWithTechnologies(
+            filterBasedOn,
+            filterValue
+          ).filter((project) =>
+            this.filterProjectsBasedOnName(project, searchValue)
+          );
+        }
+      case 'date':
+        if (!searchValue) {
+          return this.filterProjectsWithDate();
+        } else {
+          return this.filterProjectsWithDate().filter((project) =>
+            this.filterProjectsBasedOnName(project, searchValue)
+          );
+        }
+      default:
+        if (searchValue) {
+          return this.filterProjectsWithSearchValue(searchValue, 'name');
+        }
+    }
+    return this.allProjects;
+  }
+
+  // Filtering based on name field
+  private filterProjectsBasedOnName(
+    project: ProjectDetails,
+    searchValue: string
+  ): boolean {
+    return project.name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1;
+  }
+
+  // Filtering based on user entered search value
+  private filterProjectsWithSearchValue(
+    searchValue: string,
+    field: string
+  ): ProjectDetails[] {
+    return this.allProjects.filter(
+      (project: ProjectDetails | any) =>
+        project[field].toLowerCase().indexOf(searchValue.toLowerCase()) !== -1

```

Comment:
the "field" is dynamic. (field may be a name, node, angular, java, etc..) so I can not write all the code for the different technologies.

But The editor is not finding out which type for the project[field], so In my editor, showing red underline color. Actually any is not required. But the editor is showing in red. If run lint, there is no issue. 

Previously you showed some build error when you are building application right. So that's the reason I have added any type.